### PR TITLE
Plugin to sync metadata from other applications.

### PR DESCRIPTION
**Pull request by pprkut**
This is the initial version of a plugin to synchronize metadata with other applications (at some point maybe even bi-directional). It allows syncing from multiple sources, specified either in the config file or on the command line.
One example source is already implemented in the form of syncing data from amarok over its dbus interface.
I'm not picky on the name, if someone has better suggestions I'm all ears. I picked 'psync' (player sync) to make it a bit better distinguishable from mbsync than maybe a plain 'sync' would be.

### CODE REVIEW

**Comment by tomjaspers** [comment]
This looks very cool! I've been meaning to make something to sync iTunes metadata, but a general player syncing plugin is even better, and make that task easier.

My two comments:
- Wouldn't it be better if sources could register their  own `item_types` to be used by the plugin? 
- In that same line of thinking, it could be nice if the plugin could load the sources automatically, so that we plugin doesn't have to be changed when adding new sources, and avoids the `if player == u'amarok':` stuff.

Name-wise: more of a fan of the full *PlayerSync* over the *psync*

**Comment by brunal** [comment]
That's really cool!

* name-wise, I agree with @tomjaspers. You could also name it something like "metadata_sync". Explicit is good.
* source detection & loading: @tomjaspers is right (esp. the 1st point, `psync/__init__.py` should know nothing about amarok fields). Sources management already happens in `beetsplug/fetchart.py` and `beetsplyg/lyrics.py`, if you want to have a look @pprkut.

**Comment by sampsyo** [comment]
Cool indeed! Both comments above cover my thoughts too.

I wonder if a general mechanism for adding the player name as a prefix would be useful. If I were using this plugin, for example, I might not want to use the prefix at all—since I'd only sync with one player, just plain old "playcount" might be enough (along the lines of `mpdsync`).

**Comment by pprkut** [comment]
I adapted the plugin loading to be a bit more dynamic now. Alas, the item_types are still in __init__.py, mostly because I don't know how forwarding that to the lower level classes would work and after a brief chat with @sampsyo on IRC we agreed that having them in __init__.py for now is good enough.
I'm not a big fan of rather long names as I don't really want to type that much ;) Maybe something like 'metasync' could work?
And about the app prefix, I don't mind any modifications and improvements happening in that space after the plugin is in. I do think that for now it's out of scope. Data should be stored in the database with the prefix anyway (unless we establish clear specifications on how playcount/rating/etc should be handled, that all plugins need to adhere to. But that would still be out of scope for this plugin IMHO)

**Comment by sampsyo** [comment]
If this is already useful for Amarok, should we merge this now and keep evolving it in tree? I too would lean toward `playersync` as a name, but that's purely cosmetic.

**Comment by pprkut** [comment]
No objections from my side :)
Anything left to change before you want to merge it?

**Comment by sampsyo** [comment]
Hey, I forgot to ask! Could you please write the documentation for the initial version? If this doesn't seem user-facing yet, we could of course keep it in a branch for a while until it seems documentation-worthy.

**Comment by pprkut** [comment]
I think it's perfectly fine to be used by users already. I don't think any of the architectural changes left for discussion would have any notable impact on what's presented to the user at this point.
I added some documentation, please have a look. I tried to be thorough, but I might've missed something :)

File: docs/plugins/index.rst [hunk 1]
```diff
@@ -104,6 +105,7 @@ Metadata
 * :doc:`lastimport`: Collect play counts from Last.fm.
 * :doc:`lyrics`: Automatically fetch song lyrics.
 * :doc:`mbsync`: Fetch updated metadata from MusicBrainz
+* :doc:`metasync`: Fetch metadata from local or remore sources
```
Comment (tomjaspers) [inline docs/plugins/index.rst]: Typo: 'remore'

Comment (pprkut) [inline docs/plugins/index.rst]: Thanks! Fixed

**Comment by sampsyo** [comment]
Looking great. Thank you for adding documentation.

I'm going to merge this now and describe it as "experimental" in the changelog—just because I imagine the way it works might evolve as we add more player backends.

**Comment by sampsyo** [comment]
Apropos of nothing, maybe this would be a good home for what currently lives in the `lastimport` plugin.

**Pull request merged**